aiida/orm/nodes/node.py

```diff
@@ -36,6 +36,46 @@
 _NO_DEFAULT = tuple()
 
 
+class WarnWhenNotEntered:
+    """Temporary wrapper to warn when `Node.open` is called outside of a context manager."""
+
+    def __init__(self, fileobj, name):
+        self._fileobj = fileobj
+        self._name = name
+        self._was_entered = False
+
+    def _warn_if_not_entered(self, method):
+        """Fire a warning if the object wrapper has not yet been entered."""
+        if not self._was_entered:
+            msg = '`{}` used without context manager for {}. This will raise starting from `aiida-core==2.0.0`'.format(
+                method, self._name
+            )
+            warnings.warn(msg, AiidaDeprecationWarning)  # pylint: disable=no-member
+
+    def __enter__(self):
+        self._was_entered = True
+        return self._fileobj.__enter__()
+
+    def __exit__(self, *args):
+        self._fileobj.__exit__(*args)
+
+    def __getattr__(self, key):
+        if key == '_fileobj':
+            return self._fileobj
+        return getattr(self._fileobj, key)
+
+    def __del__(self):
+        self._warn_if_not_entered('del')
+
+    def read(self, *args, **kwargs):
+        self._warn_if_not_entered('read')
```

Comment:
Nice, I like that it will warn as soon as it's being used (at least for the most commonly-used methods) 👍